### PR TITLE
Improved GPS track rendering (reducing points jittering)

### PR DESCRIPTION
**Pull request by rokuz**

### CODE REVIEW

**Comment by kshalnev** [comment]
LGTM

File: drape_frontend/gps_track_renderer.cpp [hunk 1]
```diff
@@ -123,18 +125,19 @@ void GpsTrackRenderer::UpdatePoints(vector<GpsTrackPoint> const & toAdd, vector<
...
-  for (size_t i = 0; i + 1 < m_points.size(); i++)
-    len += (m_points[i + 1].m_point - m_points[i].m_point).Length();
+  if (wasChanged)
+  {
+    vector<m2::PointD> path;
+    path.reserve(m_points.size());
+    for (size_t i = 0; i < m_points.size(); i++)
+      path.push_back(m_points[i].m_point);
+    m_pointsSpline = m2::Spline(path);
+  }
```
Comment (darina) [inline drape_frontend/gps_track_renderer.cpp]: можно сразу добавлять точки в сплайн:
m_pointsSpline = m2::Spline(m_points.size());
...
m_pointsSpline.AddPoint(m_points[i].m_point);

File: drape_frontend/gps_track_renderer.cpp [hunk 2]
```diff
@@ -286,38 +261,31 @@ void GpsTrackRenderer::RenderTrack(ScreenBase const & screen, int zoomLevel,
...
+      while (!it.BeginAgain())
+      {
+        m2::PointD const pt = it.m_pos;
+        m2::RectD pointRect(pt.x - radiusMercator, pt.y - radiusMercator,
+                            pt.x + radiusMercator, pt.y + radiusMercator);
+        if (screen.ClipRect().IsIntersect(pointRect))
         {
-          lengthFromStart += dist;
-          continue;
+          dp::Color color = CalculatePointColor(static_cast<size_t>(it.GetIndex()), pt, it.GetLength(), it.GetFullLength());
```
Comment (darina) [inline drape_frontend/gps_track_renderer.cpp]: const?

**Comment by darina** [comment]
LGTM

**Pull request merged**